overlord/snapstate/snapstate_test.go

```diff
@@ -3477,6 +3477,15 @@ func (s *snapmgrTestSuite) TestDisableDoesNotEnableAgain(c *C) {
 }
 
 func (s *snapmgrTestSuite) TestEnsureRemovesVulnerableCoreSnap(c *C) {
+	// install the snapd snap so that the core snap can be removed
+	s.state.Lock()
+	snapstate.Set(s.state, "snapd", &snapstate.SnapState{
+		SnapType: "snapd",
+		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{{Revision: snap.R(1), RealName: "snapd"}}),
+		Current:  snap.R(1),
+	})
+	s.state.Unlock()
```

Comment:
something is off, you should be able to remove revisions of core as long one still exist, so this should not be needed